File: src/main/java/com/koddy/server/member/application/usecase/query/response/MenteeSimpleSearchProfile.java
package com.koddy.server.member.application.usecase.query.response;

import com.koddy.server.member.domain.model.mentee.Mentee;

public record MenteeSimpleSearchProfile(
        long id,
        String name,
        String profileImageUrl,
        String nationality,
        String interestSchool,
        String interestMajor
) {
    public static MenteeSimpleSearchProfile from(final Mentee mentee) {
        return new MenteeSimpleSearchProfile(
                mentee.getId(),
                mentee.getName(),
                mentee.getProfileImageUrl(),
                mentee.getNationality().getCode(),
                mentee.getInterest().getSchool(),
                mentee.getInterest().getMajor()
        );
    }
}

File: src/main/java/com/koddy/server/member/application/usecase/query/response/MenteeSimpleSearchProfile.kt
package com.koddy.server.member.application.usecase.query.response

import com.koddy.server.member.domain.model.mentee.Mentee

data class MenteeSimpleSearchProfile(
    val id: Long,
    val name: String,
    val profileImageUrl: String?,
    val nationality: String,
    val interestSchool: String,
    val interestMajor: String,
) {
    companion object {
        fun from(mentee: Mentee): MenteeSimpleSearchProfile =
            MenteeSimpleSearchProfile(
                id = mentee.id,
                name = mentee.name,
                profileImageUrl = mentee.profileImageUrl,
                nationality = mentee.nationality.code,
                interestSchool = mentee.interest.school,
                interestMajor = mentee.interest.major,
            )
    }
}
